### PR TITLE
Refactor KubectlExec to KubeExec

### PR DESCRIPTION
**Pull request by orenbm**
Connected to #1855

This class performs exec commands to the K8s API but it is not
using kubectl, which is the cli client of the K8s API.
Thus, the name is misleading and we should refactor it.

### CODE REVIEW

File: app/domain/authentication/authn_k8s/kube_exec.rb [hunk 1]
```diff
@@ -126,12 +126,12 @@ def on_error(err)
       private
 
       def add_websocket_event_handlers(ws_client, body, stdin)
-        kubectl = self
+        kube = self
```
Comment (abrahamko) [inline app/domain/authentication/authn_k8s/kube_exec.rb:129]: Why is this needed? this looks weird.
Can't you just call `on_open`,  `on_message` etc.?

File: spec/app/domain/authentication/authn_k8s/inject_client_cert_spec.rb [hunk 2]
```diff
@@ -215,15 +215,15 @@
     end
 
     context "when cert is being installed" do
-      let(:kubectl_exec_response) { double("MockKubeCtlExecResponse") }
+      let(:kube_exec_response) { double("MockKubeCtlExecResponse") }
```
Comment (abrahamko) [inline spec/app/domain/authentication/authn_k8s/inject_client_cert_spec.rb:218]: leftover... :)

**Review by abrahamko — APPROVED** [review]
Looks good!
Just two small things and you're good to go.

File: app/domain/authentication/authn_k8s/kube_exec.rb (hunk 1 above)
Comment (abrahamko) [reply app/domain/authentication/authn_k8s/kube_exec.rb:129]: I mean why this line is needed, not the change.
Why you need to set `self` to a var, that's my question.

Comment (orenbm) [reply app/domain/authentication/authn_k8s/kube_exec.rb:129]: no as the brackets open a new context

Comment (orenbm) [reply app/domain/authentication/authn_k8s/kube_exec.rb:129]: well, this was here before me but i guess we can use `self` instead. is this what you'd like?

Comment (abrahamko) [reply app/domain/authentication/authn_k8s/kube_exec.rb:129]: fine

**Review by abrahamko — APPROVED** [review]

**Review by abrahamko — APPROVED** [review]
